inst/templates/derivationchain/custom.js

```diff
@@ -368,15 +368,18 @@ function generateLegend(nodes) {
   var visibleProcessors = [];
   var addExternalLine = false;
   var legendContainer = $("#legendContainer");
-  var flags = [];
-  var output = [];
   $(legendContainer).html('<strong>Explanation</strong><br/><br/><span style="text-decoration:underline">Dataset Processor Types:</span> <br/>');
   
+  var flags = []; 
+  output = [];
+  l = nodes.length;
+  for(var i=0; i<l; i++) {
```

Comment:
here, in the middle i<l -- i could just use nodes.length

Reply:
Ugh, thought that was a 1. Could you change it to nodesLength maybe? Otherwise, this seems to make sense.

Reply:
yup yup!